packages/dd-trace/test/appsec/iast/context/context-plugin.spec.js

```diff
@@ -0,0 +1,258 @@
+const { TagKey } = require('../../../../src/appsec/iast/telemetry/iast-metric')
+const { storage } = require('../../../../../datadog-core')
+const { IAST_ENABLED_TAG_KEY } = require('../../../../src/appsec/iast/tags')
+
+describe('IastContextPlugin', () => {
+  let IastContextPlugin, addSub, getAndRegisterSubscription
+  let plugin
+  let acquireRequest, initializeRequestContext, releaseRequest
+  let saveIastContext, getIastContext, cleanIastContext
+  let createTransaction, removeTransaction
+  let sendVulnerabilities
+
+  beforeEach(() => {
+    addSub = sinon.stub(IastPlugin.prototype, 'addSub')
+    getAndRegisterSubscription = sinon.stub(IastPlugin.prototype, '_getAndRegisterSubscription')
+
+    acquireRequest = sinon.stub()
+    initializeRequestContext = sinon.stub()
+    releaseRequest = sinon.stub()
+
+    saveIastContext = sinon.stub()
+    getIastContext = sinon.stub()
+    cleanIastContext = sinon.stub()
+
+    createTransaction = sinon.stub()
+    removeTransaction = sinon.stub()
+
+    sendVulnerabilities = sinon.stub()
+
+    IastContextPlugin = proxyquire('../../../../src/appsec/iast/context/context-plugin', {
+      '../iast-plugin': { IastPlugin },
+      '../overhead-controller': {
+        acquireRequest,
+        initializeRequestContext,
+        releaseRequest
+      },
+      '../iast-context': {
+        saveIastContext,
+        getIastContext,
+        cleanIastContext
+      },
+      '../taint-tracking/operations': {
+        createTransaction,
+        removeTransaction
+      },
+      '../vulnerability-reporter': {
+        sendVulnerabilities
+      }
+    })
+
+    plugin = new IastContextPlugin()
+  })
+
+  afterEach(sinon.restore)
+
+  describe('startCtxOn', () => {
+    const channelName = 'start'
+    const tag = {}
+
+    it('should add a subscription to the channel', () => {
+      plugin.startCtxOn(channelName, tag)
+
+      expect(addSub).to.be.calledOnceWith(channelName)
+      expect(getAndRegisterSubscription).to.be.calledOnceWith({ channelName, tag, tagKey: TagKey.SOURCE_TYPE })
+    })
+
+    it('should call startContext when event is published', () => {
+      plugin.startCtxOn(channelName, tag)
+
+      const startContext = sinon.stub(plugin, 'startContext')
+        .returns({ isRequestAcquired: true, iastContext: {}, store: {} })
+
+      addSub.firstCall.args[1]()
+
+      expect(startContext).to.be.calledOnce
+    })
+  })
+
+  describe('finishCtxOn', () => {
+    const channelName = 'finish'
+    it('should add a subscription to the channel', () => {
+      plugin.finishCtxOn(channelName)
+
+      expect(addSub).to.be.calledOnceWith(channelName)
+    })
+
+    it('should call finishContext when event is published', () => {
+      plugin.finishCtxOn(channelName)
+
+      const startContext = sinon.stub(plugin, 'finishContext')
+        .returns({ isRequestAcquired: true, iastContext: {}, store: {} })
+
+      addSub.firstCall.args[1]()
```

Comment:
I don't understand, this is too weird.

Reply:
1) `startContext` variable is misspelled. It should be `finishContext`.

2) The `finishCtxOn(channelName)` allows to end the context when a event is published in the `channelName` channel so invoking `addSub.firstCall.args[1]()` is equivalent to invoke `dc.channel(channelName).publish()` but without using `diagnostics_channel`